database/migrations/2019_12_17_024553_add_foreign_keys.php

```diff
@@ -0,0 +1,623 @@
+<?php
+
+use App\Models\User\User;
+use App\Models\Contact\Pet;
+use App\Models\Contact\Tag;
+use App\Models\Journal\Day;
+use App\Models\User\Module;
+use App\Models\Contact\Call;
+use App\Models\Contact\Debt;
+use App\Models\Contact\Gift;
+use App\Models\Contact\Note;
+use App\Models\Contact\Task;
+use App\Models\Journal\Entry;
+use App\Models\Settings\Term;
+use App\Models\Contact\Gender;
+use App\Models\Account\ImportJob;
+use App\Models\Account\Invitation;
+use Illuminate\Support\Facades\DB;
+use App\Models\Contact\PetCategory;
+use App\Models\Instance\SpecialDate;
+use App\Models\Journal\JournalEntry;
+use Illuminate\Support\Facades\Schema;
+use App\Models\Account\ImportJobReport;
+use App\Models\Account\ActivityStatistic;
+use App\Models\Relationship\Relationship;
+use Illuminate\Database\Schema\Blueprint;
+use App\Models\Relationship\RelationshipType;
+use Illuminate\Database\Migrations\Migration;
+use App\Models\Relationship\RelationshipTypeGroup;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+class AddForeignKeys extends Migration
+{
+    private $existingAccounts;
+    private $existingUsers;
+    private $existingContacts;
+
+    /**
+     * Add foreign keys to all the tables that don't have ones.
+     * Before adding foreign keys, in all the tables, we check whether the
+     * foreign keys actually have data from the table they point to. This will
+     * ensure data integrity from now on.
+     * In order to do this, we need to parse a lot of contacts, accounts and
+     * users, as most data point to them somehow. For those 3 models, when we
+     * launch the script, we put all those ids in arrays. Then, for each
+     * foreign key, we check in the arrays if the key exists, instead of querying
+     * the database again and again. This is much faster.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::disableForeignKeyConstraints();
```

Comment:
great, but when are you calling `Schema::enableForeignKeyConstraints()`?

Reply:
Good point!